### PR TITLE
Added explanatory docs about how to use limit/offset operators.

### LINKED ISSUE #342: Explain how limit and order interact
@sras expressed confusion about how `limit` and `offset` interact.  https://github.com/tomjaguarpaw/haskell-opaleye/issues/341 It may be surprising for someone who's used to SQLs way of doing things when `OFFSET` and `LIMIT` are attached to the same `SELECT`.

We should explain this in the docs.

### PR DESCRIPTION
**Pull request by saurabhnanda**
Fixes #342 

Wasn't able to get this to build on my machine because of the following error on master:

```
Saurabhs-MBP:haskell-opaleye saurabhnanda$ stack build

Error: While constructing the build plan, the following exceptions were encountered:
       
       In the dependencies for opaleye-0.6.0.0:
           postgresql-simple-0.5.1.1 must match >=0.5.3 && <0.6 (latest applicable is 0.5.3.0)
       needed since opaleye-0.6.0.0 is a build target.
       
Plan construction failed.
```

Hope it works, though.

### CODE REVIEW

**Comment by tomjaguarpaw** [comment]
(It looks like you have `postgresql-simple-0.5.1.1` but Opaleye needs `>= 0.5.3`)

**Comment by tomjaguarpaw** [comment]
Thanks!

**Comment by saurabhnanda** [comment]
I have the following stack.yaml, and I haven't edited it: https://github.com/vacationlabs/haskell-opaleye/blob/fix/limit-offset/stack.yaml

**Comment by tomjaguarpaw** [comment]
Is there any particular reason you needed to specify the version explicitly?

**Comment by saurabhnanda** [comment]
> Is there any particular reason you needed to specify the version explicitly?

I didn't. Isn't this the exact stack file committed to the master branch?

**Comment by tomjaguarpaw** [comment]
Interesting.  I don't use Stack so I don't really know about that file.  I've opened a ticket https://github.com/tomjaguarpaw/haskell-opaleye/issues/345 to work out how to deal with it.  Thanks for the report.

File: src/Opaleye/Order.hs [hunk 1]
```diff
@@ -72,13 +72,39 @@ descNullsLast = O.order HPQ.OrderOp { HPQ.orderDirection = HPQ.OpDesc
 {- |
 Limit the results of the given query to the given maximum number of
 items.
+
+/WARNING:/ If you're planning on using limit/offset together please use
+'offset' /before/ you use 'limit'. eg:
+
+@
+limit 10 $ offset 50 yourQuery
```
Comment (phadej) [inline src/Opaleye/Order.hs]: should thee be `limitOffset 10 50` operator, to make this common use case more obvious?

Comment (tomjaguarpaw) [reply src/Opaleye/Order.hs]: Maybe.  See https://github.com/tomjaguarpaw/haskell-opaleye/issues/341#issuecomment-341110622

**Pull request merged**